logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/SendTextMessageUseCase.kt

```diff
@@ -40,16 +37,24 @@ class SendTextMessageUseCase(
                     senderClientId = currentClientId,
                     status = Message.Status.PENDING
                 )
+            }.flatMap { message ->
                 messageRepository.persistMessage(message)
-            }.flatMap {
-                messageSender.trySendingOutgoingMessageById(conversationId, generatedMessageUuid)
-            }.onFailure {
-                kaliumLogger.e("There was an error trying to send the message $it")
-                if (it is CoreFailure.Unknown) {
-                    //TODO Did I write multiplatform logging today?
-                    it.rootCause?.printStackTrace()
-                }
-            }
+                    .flatMap {
+                        messageSender.trySendingOutgoingMessageById(conversationId, generatedMessageUuid)
+                    }.flatMap {
+                        messageRepository.updateMessage(message.copy(status = Message.Status.SENT))
+                    }.onFailure {
+                        messageRepository.updateMessage(message.copy(status = Message.Status.FAILED))
+                    }
+            }.coFold(
+                { SendTextMessageResult.Failure(generatedMessageUuid) },
```

Comment:
We can't just handle this on a text message level. Too much code repetition for all the different messages.
We should treat this in the `MessageSender` so all messages are covered.

Maybe this UseCase should never fail.
It inserts into the DB and schedules the message to be sent. There's no failure.

Perhaps the naming should be something like `CreateTextMessageUseCase`.

Reply:
I guess we need a sync :P

Reply:
now I read the code totally I agree I have not seen the details of MessageSender, and it is definetly something we have to handle there